ocaml/xapi/cli_operations.ml

```diff
@@ -4340,12 +4340,23 @@ let update_upload fd printer rpc session_id params =
     let sr =
       if List.mem_assoc "sr-uuid" params
       then Client.SR.get_by_uuid rpc session_id (List.assoc "sr-uuid" params)
-      else Client.Pool.get_default_SR ~rpc ~session_id ~self:(List.hd pools)
-  in
+      else begin
+        let sr = Client.Pool.get_default_SR ~rpc ~session_id ~self:(List.hd pools) in
+        let ref_is_valid = Server_helpers.exec_with_new_task
+            ~session_id "Checking default SR validity"
+            (fun __context -> Db.is_valid_ref __context sr) in
+        if not ref_is_valid then
+          failwith "No sr-uuid parameter was given, and the pool's default SR \
+                    is not specified or invalid. Please explicitly specify the SR to use \
```

Comment:
I agree, but @mseri makes the valid observation that "...is not specified or invalid" could be read as meaning "...is not (specified or invalid)" where I'm using the parentheses in a mathematical sense.
This could be avoided by writing "...is invalid or unspecified" (or using @mseri's suggestion of "...is not specified or is invalid", but my way is slightly shorter).